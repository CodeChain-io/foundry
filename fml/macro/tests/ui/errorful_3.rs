extern crate fml_macro;

#[fml_macro::service]
pub trait MyHandle: Service {
    fn fn_1(&self, a: i32, b: String) -> String;
    fn fn_2(&self, a: i32, (a, b): (Vec<u8>, f32)) -> Box<dyn MyHandle>; // pattern in arg
}

fn main() {

}