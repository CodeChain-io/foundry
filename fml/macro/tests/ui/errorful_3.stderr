error: You must not use a pattern for the argument
 --> $DIR/errorful_3.rs:6:28
  |
6 |     fn fn_2(&self, a: i32, (a, b): (Vec<u8>, f32)) -> Box<dyn MyHandle>; // pattern in arg
  |                            ^^^^^^^^^^^^^^^^^^^^^^
